### DISCUSSION TITLE
Hide & Show container when click Button

### DISCUSSION THREAD

**Discussion by ryzam** [Budibase/budibase#3298] [Help] [answered]
Hi

I try to find out how to do simple show and hide any component (container/textbox etc) when click Button. Eg: when click button "Test", i wanted to show the container below and if clicking back it will hide/collapse. I know there is a configuration condition to control hide/show but its not really work with any on click action.

![image](https://user-images.githubusercontent.com/597759/140954496-709b0f78-1a86-42e6-93d0-fcb784bb1bc3.png)

I might suggest to have another action task called Hide/Show component

**Comment by mjashanks** [comment, marked as answer]
You shoulld use the `Update State` action, on button click to set a variable to `true`:

![image](https://user-images.githubusercontent.com/3524181/141080711-0685c73b-f0e5-4f30-9345-237bf4dff4f9.png)

Then, you can conditionally show the container based on this value:

![image](https://user-images.githubusercontent.com/3524181/141080885-aa1f5a09-552b-474c-80b5-6c56759e5b7d.png)

**Reply by ryzam** [reply]
But how do you set key name as dynamic? I need to differentiate which panel will show or hide if your put the component/element in repeater. Every Test button will have different key name and value. I try to do something like this  but the variable is not resolve to exact value

![image](https://user-images.githubusercontent.com/597759/141095661-30613c42-b24e-4792-bc0a-da474ca8ec1e.png)

![image](https://user-images.githubusercontent.com/597759/141095900-0bbbd3ae-7deb-454e-94ca-9f1f424a8487.png)

**Reply by ryzam** [reply]
Thank you i found the solution.

![image](https://user-images.githubusercontent.com/597759/141134385-14644256-e4e9-43fa-a844-5e9ac31ad339.png)

**Reply by mjashanks** [reply]
Ahh - nice one, thanks for sharing. I had a think about this earlier, but nothing came to me - so I shied away!

**Reply by ryzam** [reply]
Then for the container to hide and show base on the Test button for every repeater loop i set to this configuration. By default sub container inside repeater will hide.

![image](https://user-images.githubusercontent.com/597759/141136903-bd58f652-bd26-4bae-8637-c0ff8ee60552.png)

Actually a lot of flexibility we can do, need to explore more .... thanks guy good job
